ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -291,89 +310,136 @@ gcp_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name
     return 0;
 }
 
+// POSIX remove, not supported for cloud objects.
 static int
 gcp_remove(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, uint32_t flags)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(flags);
-
-    return 0;
+    std::cerr << "gcp_remove: object: " << name << ": file removal is not supported." << std::endl;
+    return ENOTSUP;
 }
 
+// POSIX rename, not supported for cloud objects.
 static int
 gcp_rename(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *from, const char *to,
   uint32_t flags)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(from);
-    WT_UNUSED(to);
-    WT_UNUSED(flags);
+    std::cerr << "gcp_rename: object: " << from << ": file renaming is not supported." << std::endl;
+    return ENOTSUP;
+}
+
+static int
+gcp_object_size(WT_FILE_SYSTEM *file_system, [[maybe_unused]] WT_SESSION *session, const char *name,
+  wt_off_t *sizep)
+{
+    bool exists;
+    size_t size;
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    int ret = fs->gcp_conn->object_exists(name, exists, size);
+    if (ret != 0) {
+        std::cerr << "gcp_object_size: GetObjectMetadata request to google cloud failed."
+                  << std::endl;
+        return (ret);
+    }
+    *sizep = size;
 
     return 0;
 }
 
+// Helper function to return a list of object names for the given location.
 static int
-gcp_file_size(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, wt_off_t *sizep)
+gcp_object_list_helper(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *directory,
+  const char *prefix, char ***object_list, uint32_t *count, bool list_single)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(sizep);
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    gcp_store *gcp = reinterpret_cast<gcp_file_system *>(fs)->storage_source;
+    std::vector<std::string> objects;
+    std::string complete_prefix;
 
-    return 0;
+    *count = 0;
+
+    if (directory != nullptr) {
+        complete_prefix += directory;
+        // Add a terminating '/' if one doesn't exist.
+        if (complete_prefix.length() > 1 && complete_prefix.back() != '/')
```

Comment:
This will fail to add the path separator `/` if the directory name is a single letter.
If single letter directories are not permissible consider an assert.
Otherwise, perhaps:`!complete_prefix.empty()` is more appropriate.

Reply:
updated d11e042e49654d91f7ee8bd8450515ee77e1766e